dev-libs/crypto++/patches/crypto++-5.6.5.patchset

```diff
@@ -0,0 +1,25 @@
+From ec1e84e8ecefb81f7a233f047c75446eee9dfb54 Mon Sep 17 00:00:00 2001
+From: Leorize <alaviss@users.noreply.github.com>
+Date: Sat, 16 Dec 2017 15:06:42 +0700
+Subject: Makefile: fix architecture detection
+
+
+diff --git a/GNUmakefile b/GNUmakefile
+index 2a0ff4c..ddd5422 100644
+--- a/GNUmakefile
++++ b/GNUmakefile
+@@ -15,9 +15,9 @@ LN ?= ln -sf
+ LDCONF ?= /sbin/ldconfig -n
+ UNAME := $(shell uname)
+ 
+-IS_X86 := $(shell uname -m | $(EGREP) -v "x86_64" | $(EGREP) -i -c "i.86|x86|i86")
++IS_X86 := $(shell $(CXX) $(CXXFLAGS) -dumpmachine 2>/dev/null | $(EGREP) -v "64" | $(EGREP) -i -c "i.86|x86|i86")
```

Comment:
Haiku on x86 defines "BePC" instead of x86 afaik. Check the uname -a output.

Reply:
It's more reliable this way and have a plus: it's cross-platform (as long as the said platfrom have gcc-compatible compiler). This diff was actually extracted from [upstream](https://github.com/weidai11/cryptopp/commit/a5bf9626817e96f8c8611b40fb4699202d98638b)

Reply:
True